In [14]:
import polars as pl
import numpy as np
import pandas as pd
import optuna
import pickle
import os
from catboost import CatBoostClassifier, Pool
from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_auc_score

In [15]:
# ------------------------------
# 1. Загрузка и предобработка
# ------------------------------
train = pl.read_parquet('data/train_main_features.parquet')
test = pl.read_parquet('data/test_main_features.parquet')
target = pl.read_parquet('data/train_target.parquet')

In [16]:
cat_features = [col for col in train.columns if col.startswith("cat_feature")]

def preprocess(df):
    df = df.clone()
    num_cols = [col for col in df.columns if col.startswith("num_feature")]
    for col in num_cols:
        median_val = df[col].median()
        df = df.with_columns(pl.col(col).fill_null(median_val))
    for col in cat_features:
        df = df.with_columns(pl.col(col).fill_null(-1).cast(pl.Int32))
    return df

train = preprocess(train)
test = preprocess(test)

target_cols = [col for col in target.columns if col.startswith("target")]
y_all = target[target_cols].to_pandas()
X = train.drop("customer_id").to_pandas()

In [17]:
# ------------------------------
# 2. Подвыборка для оптимизации
# ------------------------------
X_sample, _, y_sample, _ = train_test_split(
    X, y_all, test_size=0.8, random_state=42,
    stratify=y_all['target_1_1']
)
print(f"Размер выборки для оптимизации: {X_sample.shape[0]}")

X_opt_train, X_opt_val, y_opt_train, y_opt_val = train_test_split(
    X_sample, y_sample, test_size=0.2, random_state=42,
    stratify=y_sample['target_1_1']
)

Размер выборки для оптимизации: 150000


In [18]:
# ------------------------------
# 3. Функция оптимизации
# ------------------------------
def objective(trial, X_tr, y_tr, X_val, y_val, cat_features, target_name):
    params = {
        'iterations': trial.suggest_int('iterations', 100, 400, step=50),
        'learning_rate': trial.suggest_float('learning_rate', 0.02, 0.1, log=True),
        'depth': trial.suggest_int('depth', 3, 6),
        'l2_leaf_reg': trial.suggest_int('l2_leaf_reg', 1, 7),
        'random_strength': trial.suggest_int('random_strength', 1, 5),
        'bagging_temperature': trial.suggest_float('bagging_temperature', 0.5, 1.2),
        'border_count': trial.suggest_int('border_count', 32, 128, step=32),
        'min_data_in_leaf': trial.suggest_int('min_data_in_leaf', 1, 30),
        'early_stopping_rounds': 30,
        'eval_metric': 'AUC',
        'loss_function': 'Logloss',
        'verbose': False,
        'random_seed': 42,
        'task_type': 'CPU',
        'thread_count': 4
    }
    
    y_tr_target = y_tr[target_name]
    pos = y_tr_target.sum()
    neg = len(y_tr_target) - pos
    if pos > 0 and neg > 0:
        params['class_weights'] = {0: 1, 1: neg / pos}
    
    train_pool = Pool(X_tr, label=y_tr_target, cat_features=cat_features)
    val_pool = Pool(X_val, label=y_val[target_name], cat_features=cat_features)
    
    model = CatBoostClassifier(**params)
    model.fit(train_pool, eval_set=val_pool, verbose=False)
    
    pred_val = model.predict_proba(val_pool)[:, 1]
    auc = roc_auc_score(y_val[target_name], pred_val)
    return auc

In [ ]:
# ------------------------------
# 4. Разбиение на 4 группы
# ------------------------------
num_groups = 4
group_size = len(target_cols) // num_groups
groups = [target_cols[i*group_size:(i+1)*group_size] for i in range(num_groups)]
# Добавляем остаток в последнюю группу
if len(groups[-1]) < len(target_cols) - (num_groups-1)*group_size:
    groups[-1].extend(target_cols[num_groups*group_size:])
print("Группы целевых переменных:")
for i, g in enumerate(groups):
    print(f"  Группа {i+1}: {len(g)} классов")

Группы целевых переменных:
  Группа 1: 10 классов
  Группа 2: 10 классов
  Группа 3: 10 классов
  Группа 4: 11 классов


In [24]:
best_params = {}
n_trials_per_class = 6   # сокращённое количество попыток

In [25]:
#group 1
for target_name in groups[0]:
    print(f"\nОптимизация для {target_name}...")
    study = optuna.create_study(direction='maximize', sampler=optuna.samplers.TPESampler(seed=42))
    study.optimize(
        lambda trial: objective(trial, X_opt_train, y_opt_train, X_opt_val, y_opt_val,
                                cat_features, target_name),
        n_trials=n_trials_per_class,
        show_progress_bar=True
    )
    best_params[target_name] = study.best_params
    print(f"Лучший AUC: {study.best_value:.4f}, параметры: {study.best_params}")

[I 2026-03-21 22:09:04,065] A new study created in memory with name: no-name-5f4f9f6d-131f-477d-9d23-e4959b4a9d8a



Оптимизация для target_1_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:10:04,431] Trial 0 finished with value: 0.8893075053202882 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8893075053202882.
[I 2026-03-21 22:10:54,465] Trial 1 finished with value: 0.8786670907350982 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8893075053202882.
[I 2026-03-21 22:11:34,029] Trial 2 finished with value: 0.8529882789558416 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.88930

[I 2026-03-21 22:12:40,609] A new study created in memory with name: no-name-ee62a81f-40ab-4301-817f-5ae2399e67d5


[I 2026-03-21 22:12:40,608] Trial 5 finished with value: 0.8564877071975899 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8893075053202882.
Лучший AUC: 0.8893, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_1_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:13:26,108] Trial 0 finished with value: 0.8042492817573345 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8042492817573345.
[I 2026-03-21 22:13:47,893] Trial 1 finished with value: 0.7538734119830837 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8042492817573345.
[I 2026-03-21 22:14:24,945] Trial 2 finished with value: 0.769901743559035 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.804249

[I 2026-03-21 22:15:29,241] A new study created in memory with name: no-name-ad1a13c4-7014-4fa3-b7e0-6dcc17cb4828


[I 2026-03-21 22:15:29,239] Trial 5 finished with value: 0.7664936021440603 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.8050984756401509.
Лучший AUC: 0.8051, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_1_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:16:29,952] Trial 0 finished with value: 0.8432663843586518 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8432663843586518.
[I 2026-03-21 22:17:19,905] Trial 1 finished with value: 0.8406015005975304 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8432663843586518.
[I 2026-03-21 22:18:01,228] Trial 2 finished with value: 0.8265772819077359 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.84326

[I 2026-03-21 22:19:09,064] A new study created in memory with name: no-name-1db22d22-d4ce-4fad-b044-14eb57a1fa34


[I 2026-03-21 22:19:09,062] Trial 5 finished with value: 0.8255118502639018 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8432663843586518.
Лучший AUC: 0.8433, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_1_4...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:20:10,063] Trial 0 finished with value: 0.800978509712907 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.800978509712907.
[I 2026-03-21 22:21:01,190] Trial 1 finished with value: 0.7894426706684957 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.800978509712907.
[I 2026-03-21 22:21:43,197] Trial 2 finished with value: 0.7598115784988956 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.80097850

[I 2026-03-21 22:22:52,708] A new study created in memory with name: no-name-562ae808-83ef-46ef-bc21-a54a6be43b1f


[I 2026-03-21 22:22:52,706] Trial 5 finished with value: 0.7626318631133172 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.800978509712907.
Лучший AUC: 0.8010, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_1_5...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:23:20,201] Trial 0 finished with value: 0.8482638589367 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8482638589367.
[I 2026-03-21 22:23:34,994] Trial 1 finished with value: 0.8156771516087421 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8482638589367.
[I 2026-03-21 22:24:05,775] Trial 2 finished with value: 0.8279927160708667 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.8482638589367.

[I 2026-03-21 22:24:46,847] A new study created in memory with name: no-name-bed7e049-4135-4d04-a4c5-70a6f03b8949


[I 2026-03-21 22:24:46,845] Trial 5 finished with value: 0.8276618776960087 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8482638589367.
Лучший AUC: 0.8483, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_2_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:25:14,503] Trial 0 finished with value: 0.8053898793786629 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8053898793786629.
[I 2026-03-21 22:25:22,262] Trial 1 finished with value: 0.774188194840697 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8053898793786629.
[I 2026-03-21 22:25:58,622] Trial 2 finished with value: 0.7949930202779208 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.805389

[I 2026-03-21 22:27:00,267] A new study created in memory with name: no-name-5562c281-0df2-4356-9d33-d548ccc2a8cf


[I 2026-03-21 22:27:00,265] Trial 5 finished with value: 0.7930543617304002 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8053898793786629.
Лучший AUC: 0.8054, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_2_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:28:00,080] Trial 0 finished with value: 0.910006749315351 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.910006749315351.
[I 2026-03-21 22:28:52,984] Trial 1 finished with value: 0.9032407314703023 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.910006749315351.
[I 2026-03-21 22:29:35,880] Trial 2 finished with value: 0.8896283400571278 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.91000674

[I 2026-03-21 22:30:44,309] A new study created in memory with name: no-name-e080bfe4-aaf8-46ff-b97f-e6b7ab9c68e7


[I 2026-03-21 22:30:44,307] Trial 5 finished with value: 0.8897253982743896 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.910006749315351.
Лучший AUC: 0.9100, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_2_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:31:05,846] Trial 0 finished with value: 0.7700494156928213 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7700494156928213.
[I 2026-03-21 22:31:16,748] Trial 1 finished with value: 0.73019265442404 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7700494156928213.
[I 2026-03-21 22:31:41,666] Trial 2 finished with value: 0.7450260434056761 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.7700494

[I 2026-03-21 22:32:17,314] A new study created in memory with name: no-name-689da80e-4eb5-4352-918e-f162b7b0b410


[I 2026-03-21 22:32:17,312] Trial 5 finished with value: 0.7614110183639399 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.7700494156928213.
Лучший AUC: 0.7700, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_2_4...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:32:57,001] Trial 0 finished with value: 0.6748720626250528 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6748720626250528.
[I 2026-03-21 22:33:11,590] Trial 1 finished with value: 0.638456353784814 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6748720626250528.
[I 2026-03-21 22:33:48,508] Trial 2 finished with value: 0.6480612960231339 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.674872

[I 2026-03-21 22:34:19,574] A new study created in memory with name: no-name-d0aec076-63d9-4640-a1bb-81c71f94ca00


[I 2026-03-21 22:34:19,572] Trial 5 finished with value: 0.6506980327437893 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.6748720626250528.
Лучший AUC: 0.6749, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_2_5...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:34:45,571] Trial 0 finished with value: 0.6573673988222034 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6573673988222034.
[I 2026-03-21 22:34:55,377] Trial 1 finished with value: 0.6433157526185679 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6573673988222034.
[I 2026-03-21 22:35:07,855] Trial 2 finished with value: 0.6303342902057663 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.65736

In [26]:
#group 2
for target_name in groups[1]:
    print(f"\nОптимизация для {target_name}...")
    study = optuna.create_study(direction='maximize', sampler=optuna.samplers.TPESampler(seed=42))
    study.optimize(
        lambda trial: objective(trial, X_opt_train, y_opt_train, X_opt_val, y_opt_val,
                                cat_features, target_name),
        n_trials=n_trials_per_class,
        show_progress_bar=True
    )
    best_params[target_name] = study.best_params
    print(f"Лучший AUC: {study.best_value:.4f}, параметры: {study.best_params}")

[I 2026-03-21 22:36:01,905] A new study created in memory with name: no-name-f5eafa0f-3b35-4e02-8e95-24ed96377f82



Оптимизация для target_2_6...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:36:34,124] Trial 0 finished with value: 0.6925373789407423 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6925373789407423.
[I 2026-03-21 22:36:56,400] Trial 1 finished with value: 0.6350845176976074 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6925373789407423.
[I 2026-03-21 22:37:29,525] Trial 2 finished with value: 0.6321697399499882 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.69253

[I 2026-03-21 22:38:26,969] A new study created in memory with name: no-name-66f25b4f-a6f8-4a7e-b830-a725c795f3a7


[I 2026-03-21 22:38:26,968] Trial 5 finished with value: 0.6709701488227584 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.6975229702181232.
Лучший AUC: 0.6975, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_2_7...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:38:37,934] Trial 0 finished with value: 0.8013465046606113 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8013465046606113.
[I 2026-03-21 22:38:45,011] Trial 1 finished with value: 0.7928254687998628 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8013465046606113.
[I 2026-03-21 22:38:52,903] Trial 2 finished with value: 0.8297507513657949 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 2 with value: 0.82975

[I 2026-03-21 22:39:13,891] A new study created in memory with name: no-name-9d761268-43f7-4dac-b2bb-9a870308fd9c


[I 2026-03-21 22:39:13,889] Trial 5 finished with value: 0.8023824606693943 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 4 with value: 0.835499711837524.
Лучший AUC: 0.8355, параметры: {'iterations': 100, 'learning_rate': 0.09210273435223537, 'depth': 6, 'l2_leaf_reg': 6, 'random_strength': 2, 'bagging_temperature': 0.5683704798044688, 'border_count': 96, 'min_data_in_leaf': 14}

Оптимизация для target_2_8...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:39:31,673] Trial 0 finished with value: 0.9998833294443148 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.9998833294443148.
[I 2026-03-21 22:39:44,671] Trial 1 finished with value: 0.9954165138837962 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.9998833294443148.
[I 2026-03-21 22:39:58,750] Trial 2 finished with value: 0.9958998633287776 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.99988

[I 2026-03-21 22:40:37,309] A new study created in memory with name: no-name-9d8ad2eb-f7d3-4390-9a59-9108d01c3251


[I 2026-03-21 22:40:37,307] Trial 5 finished with value: 0.997233241108037 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.9998833294443148.
Лучший AUC: 0.9999, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_3_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:41:38,549] Trial 0 finished with value: 0.661213852893367 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.661213852893367.
[I 2026-03-21 22:42:28,190] Trial 1 finished with value: 0.6487624585294774 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.661213852893367.
[I 2026-03-21 22:43:09,953] Trial 2 finished with value: 0.6252376074805704 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.66121385

[I 2026-03-21 22:44:17,979] A new study created in memory with name: no-name-dd21ce42-98ee-4765-bde1-ee3d756a855f


[I 2026-03-21 22:44:17,977] Trial 5 finished with value: 0.6297024355576277 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.661213852893367.
Лучший AUC: 0.6612, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_3_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:45:17,933] Trial 0 finished with value: 0.8974333663209391 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8974333663209391.
[I 2026-03-21 22:46:07,454] Trial 1 finished with value: 0.8929264705815686 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8974333663209391.
[I 2026-03-21 22:46:48,837] Trial 2 finished with value: 0.8877119204332302 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.89743

[I 2026-03-21 22:47:54,789] A new study created in memory with name: no-name-217e2ec6-8b03-4f98-add4-4371ab85855c


[I 2026-03-21 22:47:54,786] Trial 5 finished with value: 0.8875873417253519 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8974333663209391.
Лучший AUC: 0.8974, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_3_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:48:14,850] Trial 0 finished with value: 0.7556905413523798 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7556905413523798.
[I 2026-03-21 22:48:29,246] Trial 1 finished with value: 0.7466567301037454 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7556905413523798.
[I 2026-03-21 22:48:39,746] Trial 2 finished with value: 0.7073925217556273 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.75569

[I 2026-03-21 22:49:02,836] A new study created in memory with name: no-name-a3278db9-8737-4dbf-a649-f152a4c3e607


[I 2026-03-21 22:49:02,835] Trial 5 finished with value: 0.7508422949771465 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.7556905413523798.
Лучший AUC: 0.7557, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_3_4...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:49:40,783] Trial 0 finished with value: 0.924520927793485 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.924520927793485.
[I 2026-03-21 22:49:48,058] Trial 1 finished with value: 0.8971073858871894 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.924520927793485.
[I 2026-03-21 22:50:14,107] Trial 2 finished with value: 0.9002921307736642 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.92452092

[I 2026-03-21 22:51:16,897] A new study created in memory with name: no-name-df3eaeae-595d-4369-8184-3744fa37ed77


[I 2026-03-21 22:51:16,895] Trial 5 finished with value: 0.9090705863146257 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.9270804732469061.
Лучший AUC: 0.9271, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_3_5...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:51:32,027] Trial 0 finished with value: 0.965834000800961 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.965834000800961.
[I 2026-03-21 22:51:40,797] Trial 1 finished with value: 0.9503056445512394 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.965834000800961.
[I 2026-03-21 22:51:52,223] Trial 2 finished with value: 0.9528424850561414 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.96583400

[I 2026-03-21 22:52:22,105] A new study created in memory with name: no-name-8db25acc-a804-44e2-a60e-bff6029bfafe


[I 2026-03-21 22:52:22,103] Trial 5 finished with value: 0.9529379700084546 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.965834000800961.
Лучший AUC: 0.9658, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_4_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:52:43,964] Trial 0 finished with value: 0.8519561913872464 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8519561913872464.
[I 2026-03-21 22:53:30,891] Trial 1 finished with value: 0.8521661201272391 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 1 with value: 0.8521661201272391.
[I 2026-03-21 22:54:00,336] Trial 2 finished with value: 0.8442844041722597 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 1 with value: 0.85216

[I 2026-03-21 22:54:45,473] A new study created in memory with name: no-name-226c6a77-2cd1-4e90-8f1a-0212c31b1bc1


[I 2026-03-21 22:54:45,471] Trial 5 finished with value: 0.8537470091860413 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 4 with value: 0.8569135167002759.
Лучший AUC: 0.8569, параметры: {'iterations': 100, 'learning_rate': 0.09210273435223537, 'depth': 6, 'l2_leaf_reg': 6, 'random_strength': 2, 'bagging_temperature': 0.5683704798044688, 'border_count': 96, 'min_data_in_leaf': 14}

Оптимизация для target_5_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:55:15,279] Trial 0 finished with value: 0.7296898141914636 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7296898141914636.
[I 2026-03-21 22:55:40,878] Trial 1 finished with value: 0.7198717056057959 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7296898141914636.
[I 2026-03-21 22:56:16,843] Trial 2 finished with value: 0.7250318050024809 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.72968

In [27]:
#group 3
for target_name in groups[2]:
    print(f"\nОптимизация для {target_name}...")
    study = optuna.create_study(direction='maximize', sampler=optuna.samplers.TPESampler(seed=42))
    study.optimize(
        lambda trial: objective(trial, X_opt_train, y_opt_train, X_opt_val, y_opt_val,
                                cat_features, target_name),
        n_trials=n_trials_per_class,
        show_progress_bar=True
    )
    best_params[target_name] = study.best_params
    print(f"Лучший AUC: {study.best_value:.4f}, параметры: {study.best_params}")

[I 2026-03-21 22:57:15,326] A new study created in memory with name: no-name-1e5a9b72-302d-4c89-a515-cc0b41dc6e3a



Оптимизация для target_5_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:57:43,896] Trial 0 finished with value: 0.6827209899195001 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6827209899195001.
[I 2026-03-21 22:57:56,647] Trial 1 finished with value: 0.6687506116319407 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6827209899195001.
[I 2026-03-21 22:58:04,831] Trial 2 finished with value: 0.66111039740068 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.6827209

[I 2026-03-21 22:58:26,941] A new study created in memory with name: no-name-c93508b4-a38b-4515-95c8-efb62b480c6f


[I 2026-03-21 22:58:26,939] Trial 5 finished with value: 0.712220275291761 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 5 with value: 0.712220275291761.
Лучший AUC: 0.7122, параметры: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}

Оптимизация для target_6_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 22:59:12,351] Trial 0 finished with value: 0.6942677406144715 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6942677406144715.
[I 2026-03-21 22:59:25,272] Trial 1 finished with value: 0.6553529815381351 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6942677406144715.
[I 2026-03-21 23:00:02,705] Trial 2 finished with value: 0.6720351275579102 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.69426

[I 2026-03-21 23:01:07,690] A new study created in memory with name: no-name-8ce810cb-4825-4788-bf78-428b7dfd4604


[I 2026-03-21 23:01:07,688] Trial 5 finished with value: 0.6685736411691209 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.6942677406144715.
Лучший AUC: 0.6943, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_6_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:01:39,961] Trial 0 finished with value: 0.6855579516565085 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6855579516565085.
[I 2026-03-21 23:01:47,460] Trial 1 finished with value: 0.6502409369002777 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6855579516565085.
[I 2026-03-21 23:02:08,414] Trial 2 finished with value: 0.6649456926315531 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.68555

[I 2026-03-21 23:02:50,673] A new study created in memory with name: no-name-95ee9ebf-47ae-4533-a7c9-804dd4a2f19c


[I 2026-03-21 23:02:50,672] Trial 5 finished with value: 0.6809463183624225 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.6855579516565085.
Лучший AUC: 0.6856, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_6_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:03:10,661] Trial 0 finished with value: 0.6820930267989092 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6820930267989092.
[I 2026-03-21 23:03:21,240] Trial 1 finished with value: 0.6378382111715445 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6820930267989092.
[I 2026-03-21 23:03:37,570] Trial 2 finished with value: 0.651955147249265 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.682093

[I 2026-03-21 23:04:40,754] A new study created in memory with name: no-name-6a561030-a41c-4718-a9c8-b0a756ee5a9f


[I 2026-03-21 23:04:40,751] Trial 5 finished with value: 0.6653861896999151 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.6926543463798367.
Лучший AUC: 0.6927, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_6_4...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:05:35,052] Trial 0 finished with value: 0.8240549132650683 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8240549132650683.
[I 2026-03-21 23:06:19,624] Trial 1 finished with value: 0.8222253081209359 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8240549132650683.
[I 2026-03-21 23:06:53,004] Trial 2 finished with value: 0.8158493724431666 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.82405

[I 2026-03-21 23:07:58,485] A new study created in memory with name: no-name-c634705f-808f-4744-a7a0-2d3677d08708


[I 2026-03-21 23:07:58,483] Trial 5 finished with value: 0.819018923214974 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.8294469161888063.
Лучший AUC: 0.8294, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_6_5...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:08:10,509] Trial 0 finished with value: 0.8667466398745998 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8667466398745998.
[I 2026-03-21 23:08:21,262] Trial 1 finished with value: 0.7854803812033084 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8667466398745998.
[I 2026-03-21 23:08:39,187] Trial 2 finished with value: 0.8654907617395944 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.86674

[I 2026-03-21 23:09:03,008] A new study created in memory with name: no-name-577e25f8-4456-43b2-a887-3b424d3a2036


[I 2026-03-21 23:09:03,006] Trial 5 finished with value: 0.8722599969983992 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 4 with value: 0.8925437316568837.
Лучший AUC: 0.8925, параметры: {'iterations': 100, 'learning_rate': 0.09210273435223537, 'depth': 6, 'l2_leaf_reg': 6, 'random_strength': 2, 'bagging_temperature': 0.5683704798044688, 'border_count': 96, 'min_data_in_leaf': 14}

Оптимизация для target_7_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:10:02,749] Trial 0 finished with value: 0.7799049886189618 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7799049886189618.
[I 2026-03-21 23:10:52,409] Trial 1 finished with value: 0.7706029228815272 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7799049886189618.
[I 2026-03-21 23:11:33,545] Trial 2 finished with value: 0.7553317759264826 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.77990

[I 2026-03-21 23:12:39,646] A new study created in memory with name: no-name-f2e80fda-771c-4e2a-bc5f-cc59ca53931d


[I 2026-03-21 23:12:39,644] Trial 5 finished with value: 0.7524895644047958 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.7799049886189618.
Лучший AUC: 0.7799, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_7_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:13:40,333] Trial 0 finished with value: 0.807958307747176 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.807958307747176.
[I 2026-03-21 23:14:29,960] Trial 1 finished with value: 0.7973336237216222 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.807958307747176.
[I 2026-03-21 23:15:09,808] Trial 2 finished with value: 0.7631285498501977 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.80795830

[I 2026-03-21 23:17:04,272] A new study created in memory with name: no-name-27262d63-d889-40a9-96e9-1416051cc019


[I 2026-03-21 23:17:04,269] Trial 5 finished with value: 0.767296927917828 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.807958307747176.
Лучший AUC: 0.8080, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_7_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:17:28,207] Trial 0 finished with value: 0.7151739476025145 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7151739476025145.
[I 2026-03-21 23:17:35,970] Trial 1 finished with value: 0.6647827910149772 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7151739476025145.
[I 2026-03-21 23:17:47,468] Trial 2 finished with value: 0.6674665586900874 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.71517

[I 2026-03-21 23:18:51,887] A new study created in memory with name: no-name-6d029f9f-3a84-44e2-9ca8-b02899f5e6aa


[I 2026-03-21 23:18:51,885] Trial 5 finished with value: 0.6907577163933101 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.7426294691704824.
Лучший AUC: 0.7426, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_8_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:19:53,229] Trial 0 finished with value: 0.9637273766257822 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.9637273766257822.
[I 2026-03-21 23:20:44,968] Trial 1 finished with value: 0.9611029073003421 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.9637273766257822.
[I 2026-03-21 23:21:30,302] Trial 2 finished with value: 0.9559356851789061 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.96372

In [28]:
#group 4
for target_name in groups[3]:
    print(f"\nОптимизация для {target_name}...")
    study = optuna.create_study(direction='maximize', sampler=optuna.samplers.TPESampler(seed=42))
    study.optimize(
        lambda trial: objective(trial, X_opt_train, y_opt_train, X_opt_val, y_opt_val,
                                cat_features, target_name),
        n_trials=n_trials_per_class,
        show_progress_bar=True
    )
    best_params[target_name] = study.best_params
    print(f"Лучший AUC: {study.best_value:.4f}, параметры: {study.best_params}")

[I 2026-03-21 23:27:38,547] A new study created in memory with name: no-name-5780cc65-2b54-411d-b97b-41e185d16063



Оптимизация для target_8_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:28:40,168] Trial 0 finished with value: 0.8168028471094473 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8168028471094473.
[I 2026-03-21 23:28:59,735] Trial 1 finished with value: 0.7409589722874033 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8168028471094473.
[I 2026-03-21 23:29:40,920] Trial 2 finished with value: 0.7635410042234279 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.81680

[I 2026-03-21 23:30:49,180] A new study created in memory with name: no-name-3b07613b-88ff-4fd3-ac50-2bd4d8269929


[I 2026-03-21 23:30:49,178] Trial 5 finished with value: 0.7666427212118018 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8168028471094473.
Лучший AUC: 0.8168, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_8_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:31:37,410] Trial 0 finished with value: 0.849415832440619 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.849415832440619.
[I 2026-03-21 23:32:27,821] Trial 1 finished with value: 0.8456395109157402 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.849415832440619.
[I 2026-03-21 23:33:08,711] Trial 2 finished with value: 0.8206911602822208 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.84941583

[I 2026-03-21 23:34:15,580] A new study created in memory with name: no-name-0b08591d-8cae-4462-9bb3-ed3e28a2e533


[I 2026-03-21 23:34:15,579] Trial 5 finished with value: 0.8203118318495808 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.849415832440619.
Лучший AUC: 0.8494, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_9_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:34:54,347] Trial 0 finished with value: 0.7530208359015722 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7530208359015722.
[I 2026-03-21 23:35:07,832] Trial 1 finished with value: 0.7482851502375694 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7530208359015722.
[I 2026-03-21 23:35:30,776] Trial 2 finished with value: 0.7520735226609654 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.75302

[I 2026-03-21 23:36:05,344] A new study created in memory with name: no-name-e03af97e-a5c9-4de5-8b15-759efea5163c


[I 2026-03-21 23:36:05,343] Trial 5 finished with value: 0.7505520979722947 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.7582927698353157.
Лучший AUC: 0.7583, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_9_2...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:37:05,134] Trial 0 finished with value: 0.8178718569127259 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8178718569127259.
[I 2026-03-21 23:37:54,517] Trial 1 finished with value: 0.8145135718131601 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8178718569127259.
[I 2026-03-21 23:38:34,709] Trial 2 finished with value: 0.8080139761989206 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.81787

[I 2026-03-21 23:39:40,948] A new study created in memory with name: no-name-400f692d-540c-404b-9021-850342b9bc39


[I 2026-03-21 23:39:40,947] Trial 5 finished with value: 0.8084415181490252 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.8178718569127259.
Лучший AUC: 0.8179, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_9_3...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:40:19,996] Trial 0 finished with value: 0.6613291726697934 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.6613291726697934.
[I 2026-03-21 23:41:09,197] Trial 1 finished with value: 0.6569410370646892 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.6613291726697934.
[I 2026-03-21 23:41:48,691] Trial 2 finished with value: 0.646484293305745 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.661329

[I 2026-03-21 23:42:52,637] A new study created in memory with name: no-name-57099396-a832-4ce4-ae1a-5a226ad9d69b


[I 2026-03-21 23:42:52,636] Trial 5 finished with value: 0.6481971035572924 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.6613291726697934.
Лучший AUC: 0.6613, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_9_4...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:43:11,883] Trial 0 finished with value: 0.8635822257841893 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.8635822257841893.
[I 2026-03-21 23:43:30,000] Trial 1 finished with value: 0.8626626693480364 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.8635822257841893.
[I 2026-03-21 23:43:50,189] Trial 2 finished with value: 0.8616138679252284 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.86358

[I 2026-03-21 23:44:24,107] A new study created in memory with name: no-name-28aaeb95-4fe7-4cf0-8d57-038404d645e1


[I 2026-03-21 23:44:24,105] Trial 5 finished with value: 0.8638240190367352 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 3 with value: 0.8648122175471404.
Лучший AUC: 0.8648, параметры: {'iterations': 250, 'learning_rate': 0.07076922519051033, 'depth': 3, 'l2_leaf_reg': 4, 'random_strength': 3, 'bagging_temperature': 0.5325152889039984, 'border_count': 96, 'min_data_in_leaf': 6}

Оптимизация для target_9_5...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:44:44,225] Trial 0 finished with value: 0.7900257576313079 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7900257576313079.
[I 2026-03-21 23:45:32,108] Trial 1 finished with value: 0.7955173697124567 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 1 with value: 0.7955173697124567.
[I 2026-03-21 23:45:42,524] Trial 2 finished with value: 0.7756762632030775 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 1 with value: 0.79551

[I 2026-03-21 23:46:27,088] A new study created in memory with name: no-name-f2983e58-18de-4990-acec-7eee949e0b3e


[I 2026-03-21 23:46:27,086] Trial 5 finished with value: 0.7826203953953781 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 1 with value: 0.7955173697124567.
Лучший AUC: 0.7955, параметры: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}

Оптимизация для target_9_6...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:47:28,983] Trial 0 finished with value: 0.675953762611255 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.675953762611255.
[I 2026-03-21 23:48:20,256] Trial 1 finished with value: 0.6695588475526109 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.675953762611255.
[I 2026-03-21 23:49:03,286] Trial 2 finished with value: 0.6571083458873906 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.67595376

[I 2026-03-21 23:50:11,045] A new study created in memory with name: no-name-6fbbdd91-4e03-47ff-a9f7-07e7120c77a9


[I 2026-03-21 23:50:11,043] Trial 5 finished with value: 0.65633657272522 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.675953762611255.
Лучший AUC: 0.6760, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_9_7...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:51:11,040] Trial 0 finished with value: 0.7412062518593076 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7412062518593076.
[I 2026-03-21 23:52:00,547] Trial 1 finished with value: 0.7371851903024175 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7412062518593076.
[I 2026-03-21 23:52:42,436] Trial 2 finished with value: 0.726763115415405 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.741206

[I 2026-03-21 23:53:48,339] A new study created in memory with name: no-name-4302b191-0260-403e-b457-cab8753ddd1d


[I 2026-03-21 23:53:48,337] Trial 5 finished with value: 0.7261916101446612 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 0 with value: 0.7412062518593076.
Лучший AUC: 0.7412, параметры: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}

Оптимизация для target_9_8...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:54:26,140] Trial 0 finished with value: 0.9221648084515299 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.9221648084515299.
[I 2026-03-21 23:55:15,260] Trial 1 finished with value: 0.9232495338922747 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 1 with value: 0.9232495338922747.
[I 2026-03-21 23:55:53,274] Trial 2 finished with value: 0.908874609538125 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 1 with value: 0.923249

[I 2026-03-21 23:56:57,677] A new study created in memory with name: no-name-a0c093e0-7c73-4afa-a9aa-8348dc67bfb9


[I 2026-03-21 23:56:57,675] Trial 5 finished with value: 0.9112279475874829 and parameters: {'iterations': 100, 'learning_rate': 0.04437555550097432, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 2, 'bagging_temperature': 0.9637655990477874, 'border_count': 64, 'min_data_in_leaf': 16}. Best is trial 1 with value: 0.9232495338922747.
Лучший AUC: 0.9232, параметры: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}

Оптимизация для target_10_1...


  0%|          | 0/6 [00:00<?, ?it/s]

[I 2026-03-21 23:57:56,229] Trial 0 finished with value: 0.7386493822511297 and parameters: {'iterations': 200, 'learning_rate': 0.09237421878009823, 'depth': 5, 'l2_leaf_reg': 5, 'random_strength': 1, 'bagging_temperature': 0.6091961642353418, 'border_count': 32, 'min_data_in_leaf': 26}. Best is trial 0 with value: 0.7386493822511297.
[I 2026-03-21 23:58:46,353] Trial 1 finished with value: 0.7327357845932962 and parameters: {'iterations': 300, 'learning_rate': 0.06251028636335225, 'depth': 3, 'l2_leaf_reg': 7, 'random_strength': 5, 'bagging_temperature': 0.6486373774747933, 'border_count': 32, 'min_data_in_leaf': 6}. Best is trial 0 with value: 0.7386493822511297.
[I 2026-03-21 23:59:29,345] Trial 2 finished with value: 0.7277264794471681 and parameters: {'iterations': 200, 'learning_rate': 0.04653920936389926, 'depth': 4, 'l2_leaf_reg': 3, 'random_strength': 4, 'bagging_temperature': 0.5976457024564292, 'border_count': 64, 'min_data_in_leaf': 11}. Best is trial 0 with value: 0.73864

In [31]:
best_params

{'target_1_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_2': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_1_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_4': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_5': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l

{'target_1_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_2': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_1_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_4': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_1_5': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_2_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_2_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_2_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_2_4': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_2_5': {'iterations': 100,
  'learning_rate': 0.09210273435223537,
  'depth': 6,
  'l2_leaf_reg': 6,
  'random_strength': 2,
  'bagging_temperature': 0.5683704798044688,
  'border_count': 96,
  'min_data_in_leaf': 14},
 'target_2_6': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_2_7': {'iterations': 100,
  'learning_rate': 0.09210273435223537,
  'depth': 6,
  'l2_leaf_reg': 6,
  'random_strength': 2,
  'bagging_temperature': 0.5683704798044688,
  'border_count': 96,
  'min_data_in_leaf': 14},
 'target_2_8': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_3_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_3_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_3_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_3_4': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_3_5': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_4_1': {'iterations': 100,
  'learning_rate': 0.09210273435223537,
  'depth': 6,
  'l2_leaf_reg': 6,
  'random_strength': 2,
  'bagging_temperature': 0.5683704798044688,
  'border_count': 96,
  'min_data_in_leaf': 14},
 'target_5_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_5_2': {'iterations': 100,
  'learning_rate': 0.04437555550097432,
  'depth': 3,
  'l2_leaf_reg': 7,
  'random_strength': 2,
  'bagging_temperature': 0.9637655990477874,
  'border_count': 64,
  'min_data_in_leaf': 16},
 'target_6_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_6_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_6_3': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_6_4': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_6_5': {'iterations': 100,
  'learning_rate': 0.09210273435223537,
  'depth': 6,
  'l2_leaf_reg': 6,
  'random_strength': 2,
  'bagging_temperature': 0.5683704798044688,
  'border_count': 96,
  'min_data_in_leaf': 14},
 'target_7_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_7_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_7_3': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_8_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_8_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_8_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_9_1': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_9_2': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_9_3': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_9_4': {'iterations': 250,
  'learning_rate': 0.07076922519051033,
  'depth': 3,
  'l2_leaf_reg': 4,
  'random_strength': 3,
  'bagging_temperature': 0.5325152889039984,
  'border_count': 96,
  'min_data_in_leaf': 6},
 'target_9_5': {'iterations': 300,
  'learning_rate': 0.06251028636335225,
  'depth': 3,
  'l2_leaf_reg': 7,
  'random_strength': 5,
  'bagging_temperature': 0.6486373774747933,
  'border_count': 32,
  'min_data_in_leaf': 6},
 'target_9_6': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_9_7': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26},
 'target_9_8': {'iterations': 300,
  'learning_rate': 0.06251028636335225,
  'depth': 3,
  'l2_leaf_reg': 7,
  'random_strength': 5,
  'bagging_temperature': 0.6486373774747933,
  'border_count': 32,
  'min_data_in_leaf': 6},
 'target_10_1': {'iterations': 200,
  'learning_rate': 0.09237421878009823,
  'depth': 5,
  'l2_leaf_reg': 5,
  'random_strength': 1,
  'bagging_temperature': 0.6091961642353418,
  'border_count': 32,
  'min_data_in_leaf': 26}}

In [ ]:
# ------------------------------
# 4. Обучение финальных моделей на всех данных
# ------------------------------
# Разделим все данные на train/val (80/20) для финального обучения
X_all_train, X_all_val, y_all_train, y_all_val = train_test_split(
    X, y_all, test_size=0.2, random_state=42,
    stratify=y_all['target_1_1']
)

models = {}


Обучение финальных моделей на всех данных...


In [33]:
# group 1
for target_name in groups[0]:
    print(f"Обучаем {target_name}...")
    params = best_params[target_name].copy()
    # Увеличиваем iterations для финальной модели
    if 'iterations' in params:
        params['iterations'] = int(params['iterations'] * 1.2)
    params.pop('early_stopping_rounds', None)
    
    y_tr_target = y_all_train[target_name]
    pos = y_tr_target.sum()
    neg = len(y_tr_target) - pos
    if pos > 0 and neg > 0:
        params['class_weights'] = {0: 1, 1: neg / pos}
    
    train_pool = Pool(X_all_train, label=y_tr_target, cat_features=cat_features)
    model = CatBoostClassifier(**params, verbose=False, random_seed=42)
    model.fit(train_pool)
    models[target_name] = model

Обучаем target_1_1...
Обучаем target_1_2...
Обучаем target_1_3...
Обучаем target_1_4...
Обучаем target_1_5...
Обучаем target_2_1...
Обучаем target_2_2...
Обучаем target_2_3...
Обучаем target_2_4...
Обучаем target_2_5...


In [34]:
# group 2
for target_name in groups[1]:
    print(f"Обучаем {target_name}...")
    params = best_params[target_name].copy()
    # Увеличиваем iterations для финальной модели
    if 'iterations' in params:
        params['iterations'] = int(params['iterations'] * 1.2)
    params.pop('early_stopping_rounds', None)
    
    y_tr_target = y_all_train[target_name]
    pos = y_tr_target.sum()
    neg = len(y_tr_target) - pos
    if pos > 0 and neg > 0:
        params['class_weights'] = {0: 1, 1: neg / pos}
    
    train_pool = Pool(X_all_train, label=y_tr_target, cat_features=cat_features)
    model = CatBoostClassifier(**params, verbose=False, random_seed=42)
    model.fit(train_pool)
    models[target_name] = model

Обучаем target_2_6...
Обучаем target_2_7...
Обучаем target_2_8...
Обучаем target_3_1...
Обучаем target_3_2...
Обучаем target_3_3...
Обучаем target_3_4...
Обучаем target_3_5...
Обучаем target_4_1...
Обучаем target_5_1...


In [35]:
# group 3
for target_name in groups[2]:
    print(f"Обучаем {target_name}...")
    params = best_params[target_name].copy()
    # Увеличиваем iterations для финальной модели
    if 'iterations' in params:
        params['iterations'] = int(params['iterations'] * 1.2)
    params.pop('early_stopping_rounds', None)
    
    y_tr_target = y_all_train[target_name]
    pos = y_tr_target.sum()
    neg = len(y_tr_target) - pos
    if pos > 0 and neg > 0:
        params['class_weights'] = {0: 1, 1: neg / pos}
    
    train_pool = Pool(X_all_train, label=y_tr_target, cat_features=cat_features)
    model = CatBoostClassifier(**params, verbose=False, random_seed=42)
    model.fit(train_pool)
    models[target_name] = model

Обучаем target_5_2...
Обучаем target_6_1...
Обучаем target_6_2...
Обучаем target_6_3...
Обучаем target_6_4...
Обучаем target_6_5...
Обучаем target_7_1...
Обучаем target_7_2...
Обучаем target_7_3...
Обучаем target_8_1...


In [36]:
# group 4
for target_name in groups[3]:
    print(f"Обучаем {target_name}...")
    params = best_params[target_name].copy()
    # Увеличиваем iterations для финальной модели
    if 'iterations' in params:
        params['iterations'] = int(params['iterations'] * 1.2)
    params.pop('early_stopping_rounds', None)
    
    y_tr_target = y_all_train[target_name]
    pos = y_tr_target.sum()
    neg = len(y_tr_target) - pos
    if pos > 0 and neg > 0:
        params['class_weights'] = {0: 1, 1: neg / pos}
    
    train_pool = Pool(X_all_train, label=y_tr_target, cat_features=cat_features)
    model = CatBoostClassifier(**params, verbose=False, random_seed=42)
    model.fit(train_pool)
    models[target_name] = model

Обучаем target_8_2...
Обучаем target_8_3...
Обучаем target_9_1...
Обучаем target_9_2...
Обучаем target_9_3...
Обучаем target_9_4...
Обучаем target_9_5...
Обучаем target_9_6...
Обучаем target_9_7...
Обучаем target_9_8...
Обучаем target_10_1...


In [37]:
# ------------------------------
# 5. Предсказание на тесте
# ------------------------------
print("\nФормирование предсказаний...")
X_test = test.drop("customer_id").to_pandas()
test_pool = Pool(X_test, cat_features=cat_features)

predictions = []
for target_name in target_cols:
    model = models[target_name]
    proba = model.predict_proba(test_pool)[:, 1]
    predictions.append(proba)

predictions = np.column_stack(predictions)
predict_cols = [f"predict_{col.replace('target_', '')}" for col in target_cols]
pred_df = pl.DataFrame(predictions, schema=predict_cols)


Формирование предсказаний...


In [ ]:
# ------------------------------
# 6. Сохранение сабмита
# ------------------------------
submit = pl.DataFrame({'customer_id': test['customer_id']}).hstack(pred_df)
submit.write_parquet("data/submit2.parquet")

: 